src/Interactive/Host/Interactive/Core/InteractiveHost.cs

```diff
@@ -175,36 +152,23 @@ private RemoteService TryStartProcess(string hostPath, CultureInfo culture, Canc
                     newProcessId = 0;
                 }
 
-                // sync:
-                while (!semaphore.WaitOne(_millisecondsTimeout))
-                {
-                    if (!CheckAlive(newProcess, hostPath))
-                    {
-                        return null;
-                    }
-
-                    WriteOutputInBackground(isError: false, string.Format(InteractiveHostResources.Attempt_to_connect_to_process_Sharp_0_failed_retrying, newProcessId));
-                    cancellationToken.ThrowIfCancellationRequested();
-                }
+                // (miziga) initializing pipe for replacement of remote channel
+                NamedPipeClientStream _clientStream = new NamedPipeClientStream(".", pipeName, PipeDirection.InOut, PipeOptions.Asynchronous);
+                JsonRpc jsonRpc = JsonRpc.Attach(_clientStream);
 
                 // instantiate remote service:
-                Service newService;
                 try
                 {
-                    newService = (Service)Activator.GetObject(
-                        typeof(Service),
-                        "ipc://" + remoteServerPort + "/" + Service.ServiceName);
-
                     cancellationToken.ThrowIfCancellationRequested();
-
-                    newService.Initialize(_replServiceProviderType, culture.Name);
+                    await _clientStream.ConnectAsync().ConfigureAwait(false);
+                    await jsonRpc.InvokeAsync<Task>("Initialize", _replServiceProviderType, culture.Name).ConfigureAwait(false);
```

Comment:
>InvokeAsync [](start = 34, length = 11)

Let's also pass the cancellationToken to InvokeAsync - use InvokeWithCancellationAsync.